### PR TITLE
[WNMGDS-2653] Examples to use local design system packages

### PR DESCRIPTION
**Pull request by pwolfert**
## Summary

Thanks to Yarn _workspaces_ and Lerna, the example projects now share dependencies and use local versions of the design systems. This will also keep the examples up-to-date. A good follow-up to add would be to test the examples with VRTs so we can make sure that they don't break as we update our other packages.

- Added the `examples` directory to our Yarn _workspace_ and Lerna
    - Ignore the example projects when we're doing version bumps and publishing through Lerna
    - Make sure the example project packages have unique names
    - Update yarn lockfile
- Updated dependencies in the examples
- Standardized the build system and presentation of the `react-app`, `preact-app`, and `preact-react-app` examples
- Retired the `create-react-app-typescript` example since CRA is no longer maintained (and because it is incompatible with _workspaces_) and replaced it with a newer TypeScript-enabled framework
- Upgraded our project Node version because the version we were using was no longer being maintained and because some of our new dependencies were only compatible with Node v18+

## How to test

1. `yarn clean && yarn install` in the root
2. Go through the README instructions for each of the affected examples to build and run them

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title` or [NO-TICKET] if this is unticketed work.
- [x] Selected appropriate `Type` (only one) label for this PR, if it is a breaking change, label should only be `Type: Breaking`
- [x] Selected appropriate `Impacts`, multiple can be selected.
- [x] Selected appropriate release milestone

### If this is a change to documentation:

- [ ] Checked for spelling and grammatical errors

### CODE REVIEW

File: examples/astro/src/components/react/AccordionExample.tsx [hunk 1]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 
 function AccordionExample() {
   return (
-    <div>
+    <>
```
Comment (pwolfert) [inline examples/astro/src/components/react/AccordionExample.tsx:6]: All of these were copied from the old example and modified

File: examples/preact-react-app/package.json [hunk 2]
```diff
@@ -1,22 +1,22 @@
...
   "private": true,
   "dependencies": {
-    "@cmsgov/design-system": "file:../../packages/design-system",
-    "preact": "^10.11.3"
+    "@cmsgov/design-system": "9.0.1",
+    "preact": "10.11.3"
   },
   "devDependencies": {
-    "@babel/core": "^7.20.12",
+    "@babel/core": "^7.21.4",
```
Comment (pwolfert) [inline examples/preact-react-app/package.json:10]: Now that we're sharing all the dependencies, I had to standardize a few of the dependencies so we didn't get errors caused by mismatched versions

File: examples/astro/src/layouts/Layout.astro [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	</body>
+</html>
+
+<style is:global>
+	body {
+		margin: 0;
+	}
+
+	h1, h2, h3, h4, h5, h6 {
+		margin-block-start: 1.5em !important;
```
[the remark this answers is not in the document]
Comment (pwolfert) [reply examples/astro/src/layouts/Layout.astro:39]: Updated it in [4fc6578](https://github.com/CMSgov/design-system/pull/2938/commits/4fc657864e11022a7a48e228a5caa75377132ab5)

[the remark this answers is not in the document]
Comment (zarahzachz) [reply examples/astro/src/layouts/Layout.astro:39]: i see it now lol

could we add padding-block-start: 1.5em to get around this? or does each component have a different margin size?

File: examples/astro/src/components/react/AccordionExample.tsx
Comment (zarahzachz) [inline examples/astro/src/components/react/AccordionExample.tsx:1]: I think Accordion styles are broken for me - there's a gap between the Accordion heading and body panel

File: examples/astro/src/components/react/ModalDialogExample.tsx
Comment (zarahzachz) [inline examples/astro/src/components/react/ModalDialogExample.tsx:1]: Seeing weird behavior with modal - it I open/close it more than once it jumps everything to the top of the page and seems to lock everything up (can't scroll)

File: examples/astro/src/components/react/PaginationExample.tsx
Comment (zarahzachz) [inline examples/astro/src/components/react/PaginationExample.tsx:1]: Pagination example doesn't appear to work for me

**Comment by zarahzachz** [comment]
Building cdn-html-css and saw an issue with Inline Error appearing inline with a form label. I think it hasn't been updated with the revised form error code?

**Comment by zarahzachz** [comment]
The adding machine doesn't work in cdn-web-components

File: examples/react-app/src/scripts/index.js [hunk 4]
```diff
@@ -1,33 +1,43 @@
...
+      <UsaBanner />
+      <header className="ds-base--inverse ds-u-padding-y--3">
+        <div className="ds-l-container">
+          <span className="ds-text-heading--xl">ExampleWebsite.gov</span>
+        </div>
+      </header>
+
+      <div className="ds-l-container ds-u-padding-top--2">
+        <div className="ds-u-measure--base">
+          <h1 className="ds-h1">React-app example</h1>
```
Comment (zarahzachz) [inline examples/react-app/src/scripts/index.js:21]: deprecated css

**Review by zarahzachz — COMMENTED** [review]
everything built, but saw some funny stuff here and there

File: examples/astro/src/components/react/AccordionExample.tsx
Comment (pwolfert) [reply examples/astro/src/components/react/AccordionExample.tsx:1]: Oh right, if you use `ds-content`, the accordion heading gets messed up 😬

Comment (pwolfert) [reply examples/astro/src/components/react/AccordionExample.tsx:1]: Addressed in https://github.com/CMSgov/design-system/pull/2942

File: examples/astro/src/components/react/ModalDialogExample.tsx
Comment (pwolfert) [reply examples/astro/src/components/react/ModalDialogExample.tsx:1]: Updating to the new API fixed it. I thought we had fixed the bugs with the old APIs, though, so this is a bit troubling.

**Review by zarahzachz — APPROVED** [review]

**Pull request merged**